src/Compiler/Checking/NameResolution.fs

```diff
@@ -3692,6 +3692,142 @@ let ResolveField sink ncenv nenv ad ty mp id allFields =
         ResolutionInfo.SendEntityPathToSink(sink, ncenv, nenv, ItemOccurence.UseInType, ad, resInfo, checker)
         rfref)
 
+/// Resolve a long identifier representing a nested record field
+let ResolveNestedField sink (ncenv: NameResolver) nenv ad recdTy lid =
```

Comment:
@kerams If it's not a big chunk of work, for future readers, could you please describe in more details what this does? Just a high-level.